Subject: [PATCH] libstagefright: use 64-bit usage for native_window_set_usage

Since Android 8.1 native_window_set_usage has changed to use 64-bit int
for the usage parameter. Use of 32-bit signed integer as parameter does
implicit conversion which may cause unexpected behaviour.

diff --git a/media/libstagefright/SurfaceUtils.cpp b/media/libstagefright/SurfaceUtils.cpp
index b7c1598..2dd6df8 100644
--- a/media/libstagefright/SurfaceUtils.cpp
+++ b/media/libstagefright/SurfaceUtils.cpp
@@ -108,9 +108,8 @@
         }
     }
 
-    uint64_t finalUsage = (usage | consumerUsage) & 0xffffffffLL;
-    ALOGV("gralloc usage: %#x(producer) + %#x(consumer) = %#" PRIx64,
-            usage, consumerUsage, finalUsage);
+    int finalUsage = usage | consumerUsage;
+    ALOGV("gralloc usage: %#x(producer) + %#x(consumer) = %#x", usage, consumerUsage, finalUsage);
     err = native_window_set_usage(nativeWindow, finalUsage);
     if (err != NO_ERROR) {
         ALOGE("native_window_set_usage failed: %s (%d)", strerror(-err), -err);
@@ -124,7 +123,7 @@
         return err;
     }
 
-    ALOGD("set up nativeWindow %p for %dx%d, color %#x, rotation %d, usage %#" PRIx64,
+    ALOGD("set up nativeWindow %p for %dx%d, color %#x, rotation %d, usage %#x",
             nativeWindow, width, height, format, rotation, finalUsage);
     return NO_ERROR;
 }
